### PR TITLE
fix: address a number of typos

### PR DESCRIPTION
**Pull request by G-Rath**
I found these while working on other stuff

### CODE REVIEW

**Review by another-rex — APPROVED** [review]

File: cmd/osv-scanner/fix/main.go [hunk 1]
```diff
@@ -274,7 +274,7 @@ func action(ctx *cli.Context, stdout, stderr io.Writer) (reporter.Reporter, erro
 			return nil, err
 		}
 		opts.ManifestRW = rw
-		// Prefer the manifest's system over the lockfile's.
```
Comment (cuixq) [inline cmd/osv-scanner/fix/main.go:277]: I think this is correct?

File: internal/thirdparty/xml/read.go
Comment (cuixq) [inline internal/thirdparty/xml/read.go:1]: this is forked from Go so I would prefer to have it not changed, so we are able to diff it in the future.

File: internal/thirdparty/xml/typeinfo.go
Comment (cuixq) [inline internal/thirdparty/xml/typeinfo.go:1]: same here

**Review by hogo6002 — APPROVED** [review]

**Review by cuixq — APPROVED** [review]

**Comment by cuixq** [comment]
@G-Rath can you fix the lint issue so I can merge this PR?

**Comment by G-Rath** [comment]
@cuixq the lint failure is a very weird fluke - it doesn't happen locally and I didn't change anything related to that file; hopefully it'll pass on this rerun

**Comment by G-Rath** [comment]
@cuixq looks like this is just a general failure on `main` - I can look into why that's happening shortly, but in the meantime you should be fine to merge this:

image

**Comment by cuixq** [comment]
yes I also noticed that - maybe this is an issue of the linter itself

**Comment by G-Rath** [comment]
@cuixq CI is green now so technically this is good to merge - I couldn't actually find an answer to why linting was failing but I'm hoping it's a very rare fluke; I suspect it could be caching related.

**Comment by cuixq** [comment]
@G-Rath thank you! I am also thinking to upgrade the linter...

**Comment by G-Rath** [comment]
ha yah I saw your PR - I was thinking of doing the exact same thing, but thankfully saw your PR first 😄

**Pull request merged**